compiler/rustc_type_ir/src/generic_arg.rs

```diff
@@ -0,0 +1,18 @@
+use rustc_macros::{HashStable_NoContext, TyDecodable, TyEncodable};
```

Comment:
feel like this doesn't necessarily deserve it's own module 🤔 would maybe also put it into `lib.rs`

*edit: we actually already have a lot of modules in `rustc_type_ir`, so let's keep the current setup, at least for this PR and maybe merge these modules separately*

Reply:
Yeah, idk, because there's also value with not creating one module that has a lot of stuff, so I am biased to split it up more. Maybe worth grouping this and other things in a different way tho.